src/Common/src/System/Runtime/InteropServices/FixedString.cs

```diff
@@ -0,0 +1,71 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.IO
+{
+    internal static class FixedString
+    {
+        /// <summary>
+        /// Returns a string from the given span, terminating the string at null if present.
+        /// </summary>
+        internal unsafe static string GetString(this Span<char> span)
+        {
+            int stringLength = span.GetStringLength();
+            if (stringLength == 0)
+                return string.Empty;
+
+            fixed (char* c = &span[0])
```

Comment:
You can use `fixed (char* c = &span.DangerousGetPinnableReference())`. The check for zero length above is not necessary with it, and it should be otherwise a tiny bit faster.